modules/gapi/include/opencv2/gapi/rmat.hpp

```diff
@@ -46,13 +46,23 @@ class RMat
 
         View() = default;
         View(const GMatDesc& desc, uchar* data, size_t step = 0u, DestroyCallback&& cb = nullptr)
-            : m_desc(desc), m_data(data), m_step(step == 0u ? elemSize()*cols() : step), m_cb(cb)
+            : m_desc(desc), m_data(data), m_step(step == 0u ? elemSize()*cols() : step), m_cb(std::move(cb))
         {}
 
         View(const View&) = delete;
-        View(View&&) = default;
         View& operator=(const View&) = delete;
-        View& operator=(View&&) = default;
+        View(View&&) = default;
+        View& operator=(View&& v) {
+            m_desc = v.m_desc;
+            m_data = v.m_data;
+            m_step = v.m_step;
+            m_cb   = v.m_cb;
+            v.m_desc = {};
+            v.m_data = nullptr;
+            v.m_step = 0u;
+            v.m_cb   = nullptr;
+            return *this;
```

Comment:
if you realy need this move constructor implicitly defined - then this probably be a bit shorter : 
```c++
auto bind =[](View& v) { return std::tie(v.m_desc, v.m_data, v.m_step, v.m_cb);}

bind(*this) = std::move(bind(v));
v = {};
return *this;
```

Reply:
Clang complains with these warnings (gcc is fine, but gcc generates correct default implementation as well):
```
warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
bind(*this) = std::move(bind(view));
              ^
note: remove std::move call here
bind(*this) = std::move(bind(view));
```
Also v = {} requires operator=(View&&), we're already implementing this method